### ISSUE TITLE
using wrong password results in a stale mount point

### ISSUE THREAD

**Issue by tessus** [libfuse/sshfs#34]
I'm using git master. Here's the output of `sshfs --version`:
```
SSHFS version 2.8
OSXFUSE 3.5.3
FUSE library version: 2.9.7
fuse: no mount point
```
 I've tried on OSX 10.9.5 and OSX 10.11.6, but I doubt it has anything to do with the OS.
Here are my steps (entering 3 times the wrong password):
```
$ ./sshfs user@server: ~/mytest
user@server's password:
user@server's password:
user@server's password:
remote host has disconnected
```
But when I check my mount points:
```
$ mount
user@server: on /Users/kct/mytest (osxfuse, nodev, nosuid, synchronous, mounted by kct) 
```
Because of this, any subsequent try to mount it again, abends with:
```
mount_osxfuse: mount point /Users/kct/mytest is itself on a OSXFUSE volume
fuse: failed to mount file system: Invalid argument
```

Please note:
I also have a very old static version of sshfs (the one that comes with Macfusion, which is sshfs 1.8) and the problem does not occur there.

**Comment by tessus** [comment]
checked the code:

```
4155		res = ssh_connect();
4156		if (res == -1) {
4157			fuse_unmount(mountpoint, ch);
4158			fuse_destroy(fuse);
4159			exit(1);
4160		}
```

It looks like `fuse_unmount` does not do its job properly.

**Comment by Nikratio** [comment]
Thanks for the report! `fuse_unmount` is provided by FUSE4X, so it doesn't work right then sshfs can't do much about it... Could you report this issue to the FUSE4X developers? Sorry that I can't be more helpful...

**Comment by tessus** [comment]
For reference:

This issue has been fixed in **FUSE for macOS 3.6.1**
